src/main/java/collectionsmaps/FileNavigator.java

```diff
@@ -0,0 +1,60 @@
+package collectionsmaps;
+
+import java.util.*;
+
+public class FileNavigator {
+    private HashMap<String, ArrayList<FileData>> files = new HashMap<String, ArrayList<FileData>>();
+
+    public HashMap<String, ArrayList<FileData>> add(String path, FileData fileData) {
+        if (!path.equals(fileData.getPathToFile()))
+            throw new PathNotEquasException("Має бути помилка, оскільки шлях-ключ і шлях до файлу не співпадають.\n" +
+                    "Шлях: " + path + "\n" +
+                    "FileData: {name: " + fileData.getFileName() + ", size: " + fileData.getSizeInBytes() + ", path: " + fileData.getPathToFile() + "}");
+        if (files.containsKey(path)) {
+            ArrayList<FileData> oldFiles = files.get(path);
+            oldFiles.add(fileData);
+            files.put(path, oldFiles);
+        } else {
+            ArrayList<FileData> fileDatas = new ArrayList<FileData>();
+            fileDatas.add(fileData);
+            files.put(path, fileDatas);
+        }
+        return files;
+    }
+
+    public ArrayList<FileData> find(String path) {
+        if (files.containsKey(path))
```

Comment:
```suggestion
        files.getOrDefault(path, List.of()); 
```
але тут є важливий момент - не загубити посилання, на уроці поговоримо.